File: app/src/main/java/com/gifola/ContactUsActivity.java
package com.gifola;

import androidx.appcompat.app.AppCompatActivity;
import androidx.appcompat.widget.Toolbar;
import androidx.cardview.widget.CardView;

import android.content.Intent;
import android.os.Bundle;
import android.view.View;
import android.widget.TextView;

public class ContactUsActivity extends AppCompatActivity {
    CardView sendcard;
    Toolbar toolbar;
    TextView txt_title;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_contact_us);
        setupToolbar();
        sendcard=findViewById(R.id.sendcard);
        sendcard.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                Intent intent = new Intent(getApplicationContext(), DashboardActivity.class);
                intent.setFlags(Intent.FLAG_ACTIVITY_CLEAR_TOP);
                startActivity(intent);
            }
        });
    }

    private void setupToolbar() {

        toolbar = (Toolbar) findViewById(R.id.toolbarsignup);
        txt_title = (TextView) findViewById(R.id.txt_title);

        setSupportActionBar(toolbar);

        txt_title.setText("Contact Us");

        getSupportActionBar().setDisplayHomeAsUpEnabled(true);
        getSupportActionBar().setHomeButtonEnabled(true);

        toolbar.setNavigationIcon(R.drawable.back_1);

        toolbar.setNavigationOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                onBackPressed();
            }
        });
    }
}

File: app/src/main/java/com/gifola/ContactUsActivity.kt
package com.gifola

import android.app.ProgressDialog
import android.content.Intent
import android.database.Cursor
import android.os.Bundle
import android.provider.ContactsContract
import android.util.Log
import android.util.Patterns
import androidx.appcompat.app.AppCompatActivity
import androidx.fragment.app.FragmentActivity
import com.gifola.apis.AdminAPI
import com.gifola.apis.SeriveGenerator
import com.gifola.constans.Global
import com.gifola.constans.SharedPreferenceHelper
import com.gifola.model.FavLocationModel
import com.gifola.model.UserData
import com.google.gson.JsonObject
import kotlinx.android.synthetic.main.activity_contact_us.*
import kotlinx.android.synthetic.main.custom_toolbar_back.*
import okhttp3.ResponseBody
import retrofit2.Call
import retrofit2.Callback
import retrofit2.Response


class ContactUsActivity : AppCompatActivity() {

    var appPreference: SharedPreferenceHelper? = null
    var adminAPI: AdminAPI? = null
    var bar: ProgressDialog? = null
    var userData: UserData? = null

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_contact_us)
        appPreference = SharedPreferenceHelper(applicationContext)
        bar = ProgressDialog(this)
        adminAPI = SeriveGenerator.getAPIClass()

        try {
            userData = Global.getUserMe(appPreference!!)
        } catch (e: Exception) {

        }


        sendcard.setOnClickListener {
            val subject = edit_subject.text.toString().trim()
            val message = edit_message.text.toString().trim()
            val emailId = edit_to.text.toString().trim()

            if (emailId.isEmpty() || !Patterns.EMAIL_ADDRESS.matcher(emailId).matches())
                Global.displayToastMessage(getString(R.string.message_valid_email_id), applicationContext)
            else if (subject.isEmpty())
                Global.displayToastMessage(getString(R.string.message_valid_subject), applicationContext)
            else if (message.isEmpty())
                Global.displayToastMessage(getString(R.string.message_valid_message), applicationContext)
            else
                sendEmailInSupport(subject, message, emailId)


        }

        setupToolbar()
    }

    private fun sendEmailInSupport(subject: String, message: String, emailId: String) {
        val jsonRequest = JsonObject();
        jsonRequest.addProperty("AppUserId", userData?.app_usr_id ?: 0)
        jsonRequest.addProperty("MobileNo", userData?.app_usr_mobile ?: "")
        jsonRequest.addProperty("To", emailId)
        jsonRequest.addProperty("Subject", subject)
        jsonRequest.addProperty("Message", message)
        Log.e("request", jsonRequest.toString())
        val responseBodyCall: Call<ResponseBody>? = adminAPI?.SendContactRequestForm(jsonRequest)
        responseBodyCall?.enqueue(object : Callback<ResponseBody?> {
            override fun onResponse(call: Call<ResponseBody?>, response: Response<ResponseBody?>) {
                if (response.code() == 200) {
                    Global.displayToastMessage(getString(R.string.message_email_sent), applicationContext)
                    val intent = Intent(this@ContactUsActivity, DashboardActivity::class.java)
                    startActivity(intent)
                    finish()
                }

            }

            override fun onFailure(call: Call<ResponseBody?>, t: Throwable) {
            }
        })
    }

    private fun setupToolbar() {
        setSupportActionBar(toolbarsignup)
        txt_title!!.text = "Contact Us"
        supportActionBar!!.setDisplayHomeAsUpEnabled(true)
        supportActionBar!!.setHomeButtonEnabled(true)
        toolbarsignup!!.setNavigationIcon(R.drawable.back_1)
        toolbarsignup!!.setNavigationOnClickListener { onBackPressed() }
    }
}